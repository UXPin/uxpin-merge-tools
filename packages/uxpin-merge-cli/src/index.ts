import 'source-map-support/register';
export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';
export { buildDesignSystemWithStorybook } from './steps/building/buildDesignSystemWithStorybook';
