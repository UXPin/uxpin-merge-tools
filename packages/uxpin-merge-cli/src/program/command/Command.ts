export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  DELETE_VERSION = 'delete-version',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND: Command = Command.EXPERIMENT;
