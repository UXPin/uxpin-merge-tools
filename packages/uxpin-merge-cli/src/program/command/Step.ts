import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun: boolean;
  exec: (infos: DSMetadata) => any;
}

export type StepExecutor = (designSystem: DSMetadata) => Promise<DSMetadata>;

export type StepWithoutDSExecutor = () => Promise<void>;
