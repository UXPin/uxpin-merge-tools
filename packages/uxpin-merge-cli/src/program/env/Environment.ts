export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  PRODUCTION = 'production',
}

export const DEFAULT_ENVIRONMENT: Environment = Environment.PRODUCTION;
