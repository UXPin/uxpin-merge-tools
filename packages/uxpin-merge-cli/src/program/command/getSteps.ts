import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getDeleteVersionCommandSteps } from './delete_version/getDeleteVersionCommandSteps';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getExperimentationCommandSteps } from './experimentation/getExperimentationCommandSteps';
import { getGeneratePresetsCommandSteps } from './generate_presets/getGeneratePresetsCommandSteps';
import { getInitCommandSteps } from './init/getInitCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args: ProgramArgs): Step[] {
  switch (args.command) {
    case Command.DUMP:
      return getDumpCommandSteps();

    case Command.EXPERIMENT:
      return getExperimentationCommandSteps(args);

    case Command.INIT:
      return getInitCommandSteps(args);

    case Command.GENERATE_PRESETS:
      return getGeneratePresetsCommandSteps(args);

    case Command.PUSH:
      return getPushCommandSteps(args);

    case Command.DELETE_VERSION:
      return getDeleteVersionCommandSteps(args);

    case Command.SERVER:
      return getServerCommandSteps(args);

    case Command.SUMMARY:
      return getSummaryCommandSteps();

    default:
      return [];
  }
}
