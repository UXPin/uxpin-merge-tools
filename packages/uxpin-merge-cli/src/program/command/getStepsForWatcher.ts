import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args: ProgramArgs): Step[] {
  if (args.command === Command.EXPERIMENT) {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
