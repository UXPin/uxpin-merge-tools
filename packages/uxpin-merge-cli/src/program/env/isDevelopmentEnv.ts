import { Environment } from './Environment';

export function isDevelopmentEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.DEVELOPMENT;
}
