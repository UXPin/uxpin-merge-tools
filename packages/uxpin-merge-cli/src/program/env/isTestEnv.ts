import { Environment } from './Environment';

export function isTestEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.TEST;
}
