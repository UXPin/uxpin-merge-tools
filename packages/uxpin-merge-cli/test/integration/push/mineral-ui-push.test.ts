import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { mineralUiServerStub } from '../../resources/stubs/mineralUi';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';

const CURRENT_TIMEOUT = 75000;

setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('Pushing mineral-ui design system', () => {
  const { getTlsPort } = setupStubbyServer(mineralUiServerStub);

  describe('with required user webpack config', () => {
    let consoleOutput: string;

    it('prints warnings without stack traces to the console', async () => {
      const params: string[] = [
        Command.PUSH,
        '--webpack-config "./webpack.config.js"',
        '--wrapper "./src/library/themes/UXPinWrapper.js"',
        '--token DUMMY_TOKEN',
      ];

      consoleOutput = await runUXPinMergeCommand({
        cwd: 'resources/repos/mineral-ui',
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params,
      });

      // then
      expect(consoleOutput).toMatchSnapshot();
    });
  });

  describe('without required user webpack config', () => {
    it('throws an error', async () => {
      const params: string[] = [Command.PUSH];

      try {
        await runUXPinMergeCommand({
          cwd: 'resources/repos/mineral-ui',
          env: {
            UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
            UXPIN_ENV: Environment.TEST,
          },
          params,
        });
      } catch (error) {
        expect(error.stderr).toMatch('Module parse failed: Unexpected token');
      }
    });
  });
});
