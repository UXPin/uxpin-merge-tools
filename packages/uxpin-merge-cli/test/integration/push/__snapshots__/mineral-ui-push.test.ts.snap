// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pushing mineral-ui design system with required user webpack config prints warnings without stack traces to the console 1`] = `
"[33mwarning [39mCannot compute default value for property \`triggerTitle\`

SyntaxError: Unexpected token ':'

 in:
[34msrc/library/Card/CardFooter/CardFooter.js[39m

[31mðŸ›‘ The current commit is not on branch [master], please specify --branch to use a custom branch
[0m"
`;
